rotkehlchen/chain/evm/decoding/curve/curve_cache.py

```diff
@@ -390,59 +410,100 @@ def _query_curve_data_from_chain(
         ))
-    except DeserializationError as e:
-        log.error(f'Curve address provider returned an invalid address for metaregistry. {e}')
-        return []
+    except (RemoteError, DeserializationError) as e:
+        log.error(
+            f'Failed to retrieve metaregistry address from the Curve '
+            f'address provider on {evm_inquirer.chain_name} due to {e!s}',
+        )
+        return
 
     metaregistry = EvmContract(
         address=metaregistry_address,
         abi=evm_inquirer.contracts.abi('CURVE_METAREGISTRY'),  # type: ignore[call-overload]  # for some reason mypy doesn't properly see argument type
         deployed_block=0,  # deployment_block is not used and the contract is dynamic
     )
-    pool_count = metaregistry.call(node_inquirer=evm_inquirer, method_name='pool_count')
-    new_pools: list[CurvePoolData] = []
+    try:
+        pool_count = metaregistry.call(node_inquirer=evm_inquirer, method_name='pool_count')
+    except RemoteError as e:
+        log.error(
+            'Failed to retrieve Curve pool count from the metaregistry '
+            f'on {evm_inquirer.chain_name} due to {e!s}',
+        )
+        return
+
+    if (existing_pool_count := len(existing_pools)) == pool_count:
+        return
+    if (new_pool_count := pool_count - existing_pool_count) > MAX_ONCHAIN_POOLS_QUERY:
+        pool_count = existing_pool_count + MAX_ONCHAIN_POOLS_QUERY
+        new_pool_count = MAX_ONCHAIN_POOLS_QUERY
+        log.error(
+            f'Tried to query {new_pool_count} {evm_inquirer.chain_name} Curve pools. '
+            f'Too many pools to query onchain. Only querying {new_pool_count}.',
+        )
+
     last_notified_ts = Timestamp(0)
-    for pool_index in range(pool_count):
-        last_notified_ts = maybe_notify_new_pools_status(
+    for pool_index in range((start_idx := pool_count - new_pool_count), pool_count):
+        last_notified_ts = maybe_notify_cache_query_status(
             msg_aggregator=msg_aggregator,
             last_notified_ts=last_notified_ts,
             protocol=CPT_CURVE,
             chain=evm_inquirer.chain_id,
-            get_new_pools_count=lambda: len(new_pools),
+            processed=(processed := pool_index - start_idx),
+            total=new_pool_count,
         )
 
-        raw_address = metaregistry.call(
-            node_inquirer=evm_inquirer,
-            method_name='pool_list',
-            arguments=[pool_index],
-        )
         try:
-            pool_address = deserialize_evm_address(raw_address)
-        except DeserializationError as e:
-            log.error(f'Could not deserialize curve pool address {raw_address}. {e}')
-            continue
+            if (pool_address := deserialize_evm_address(metaregistry.call(
+                node_inquirer=evm_inquirer,
+                method_name='pool_list',
+                arguments=[pool_index],
+            ))) in IGNORED_CURVE_POOLS | existing_pools:
+                continue
```

Comment:
mmmm  this union of sets you don't need to do it in every loop. Do it once outside